taggui/models/image_list_model.py

```diff
@@ -465,27 +482,38 @@ def add_tags(self, tags: list[str], image_indices: list[QModelIndex]):
 
     @Slot(list, str)
     def rename_tags(self, old_tags: list[str], new_tag: str,
-                    scope: Scope | str = Scope.ALL_IMAGES):
+                    scope: Scope | str = Scope.ALL_IMAGES,
+                    use_regex: bool = False):
         self.add_to_undo_stack(
             action_name=f'Rename {pluralize("Tag", len(old_tags))}',
             should_ask_for_confirmation=True)
         changed_image_indices = []
         for image_index, image in enumerate(self.images):
             if not self.is_image_in_scope(scope, image_index, image):
                 continue
-            if not any(old_tag in image.tags for old_tag in old_tags):
-                continue
+            if use_regex:
+                pattern = old_tags[0]
```

Comment:
`old_tags` is always a list containing a single item (the regex pattern) when `use_regex = True`.